client/components/editor/index.vue

```diff
@@ -36,16 +48,79 @@ export default {
   },
   data: () => ({
     isLoading: true,
-    selectedElement: null
+    selectedElement: null,
+    publishedRevisions: {}
   }),
   computed: {
+    ...mapState('editor', ['isPublishedPreview']),
     ...mapGetters('repository/userTracking', ['getActiveUsers']),
     ...mapGetters('repository', ['repository', 'outlineActivities']),
     ...mapGetters('editor', ['activity', 'contentContainers', 'rootContainerGroups']),
-    activeUsers: vm => vm.getActiveUsers('activity', vm.activityId)
+    ...mapGetters('repository/contentElements', { allElements: 'elements' }),
+    ...mapChannels({ editorChannel: 'editor' }),
+    activeUsers: vm => vm.getActiveUsers('activity', vm.activityId),
+    containerIds: vm => vm.contentContainers.map(it => it.id),
+    elements() {
+      const elements = pickBy(this.allElements, this.isActivityElement);
+      return mapValues(elements, this.addPublishFlags);
+    },
+    changesSincePublish() {
+      const { elements, publishedRevisions } = this;
+      return this.unionElementsAndRevisions(elements, publishedRevisions);
+    }
   },
   methods: {
+    ...mapMutations('editor', ['setIsPublishedPreview']),
     ...mapActions('repository', ['initialize']),
+    unionElementsAndRevisions(elements, revisions) {
```

Comment:
There is too much preview related logic in this core view, we need to extract it / handle it somewhere else.